web/controllers/user_group_controller.ex

```diff
@@ -10,6 +10,66 @@ defmodule Pairmotron.UserGroupController do
   alias Pairmotron.UserGroup
   import Pairmotron.ControllerHelpers
 
+  @spec edit(Plug.Conn.t, map()) :: Plug.Conn.t
+  def edit(conn, %{"group_id" => group_id, "user_id" => user_id}) do
+    user_group = user_id |> UserGroup.user_group_for_user_and_group(group_id) |> Repo.one
+    user = conn.assigns.current_user
+    logged_in_users_user_group = user.id |> UserGroup.user_group_for_user_and_group(group_id) |> Repo.one
+
+    cond do
+      is_nil(user_group) ->
+        redirect_not_authorized(conn, group_path(conn, :show, group_id))
+      is_nil(logged_in_users_user_group) ->
+        redirect_not_authorized(conn, group_path(conn, :show, group_id))
+      user_group.group.owner_id == user.id ->
+        render_edit(conn, user_group)
+      logged_in_users_user_group.is_admin ->
+        render_edit(conn, user_group)
+      true ->
+        redirect_not_authorized(conn, group_path(conn, :show, group_id))
+    end
```

Comment:
Yep! There's probably a better way to do this utilizing `with`, but that's a thought for a different day I think. I've been thinking of refactoring all controllers to use with and just use a case at the end to prepare for the eventual use of a FallBackController, since that structure would make the coming code migration a lot easier.